addons/mail/controllers/discuss/channel.py

```diff
@@ -160,7 +160,8 @@ def discuss_channel_fold(self, channel_id, state, state_count):
     @http.route("/discuss/channel/join", methods=["POST"], type="jsonrpc", auth="public")
     @add_guest_to_context
     def discuss_channel_join(self, channel_id):
-        channel = request.env["discuss.channel"].search([("id", "=", channel_id)])
+        # sudo(): add a guest as a channel member.
+        channel = request.env["discuss.channel"].sudo().search([("id", "=", channel_id)])
```

Comment:
Why is this needed now? That does not look acceptable *at all* since now anyone can join any channel. e.g. with this scenario that was not possible before:
- access a public channel with a guest => guest is created, joins the public channel => fine for now
- execute `odoo.__WOWL_DEBUG__.root.env.services.orm.rpc("/discuss/channel/join", { channel_id: 2 })` in the console => guest user just joined the administrator channel.